src/libraries/System.Text.Json/tests/Common/RequiredKeywordTests.cs

```diff
@@ -143,6 +144,85 @@ public async Task ClassWithRequiredKeywordAndSmallParametrizedCtorFailsDeseriali
+            Assert.Equal(3, typeInfo.Properties.Count);
+
+            AssertJsonTypeInfoHasRequiredProperties(GetTypeInfo<InheritedPersonWithRequiredMembers>(options),
+                nameof(InheritedPersonWithRequiredMembers.FirstName),
+                nameof(InheritedPersonWithRequiredMembers.LastName));
+
+            JsonException exception = await Assert.ThrowsAsync<JsonException>(() => Serializer.DeserializeWrapper("{}", typeInfo));
+            Assert.Contains("FirstName", exception.Message);
+            Assert.Contains("LastName", exception.Message);
+            Assert.DoesNotContain("MiddleName", exception.Message);
+        }
+
+        [Fact]
+        public async Task InheritedPersonWithRequiredMembersWithAdditionalRequiredMembersWorksAsExpected()
+        {
+            var options = new JsonSerializerOptions(Serializer.DefaultOptions);
+            options.MakeReadOnly();
+
+            JsonTypeInfo typeInfo = options.GetTypeInfo(typeof(InheritedPersonWithRequiredMembersWithAdditionalRequiredMembers));
+            Assert.Equal(4, typeInfo.Properties.Count);
+
+            AssertJsonTypeInfoHasRequiredProperties(GetTypeInfo<InheritedPersonWithRequiredMembersWithAdditionalRequiredMembers>(options),
+                nameof(InheritedPersonWithRequiredMembersWithAdditionalRequiredMembers.FirstName),
+                nameof(InheritedPersonWithRequiredMembersWithAdditionalRequiredMembers.LastName),
+                nameof(InheritedPersonWithRequiredMembersWithAdditionalRequiredMembers.Post));
+
+            JsonException exception = await Assert.ThrowsAsync<JsonException>(() => Serializer.DeserializeWrapper("{}", typeInfo));
+            Assert.Contains("FirstName", exception.Message);
+            Assert.Contains("LastName", exception.Message);
+            Assert.Contains("Post", exception.Message);
+            Assert.DoesNotContain("MiddleName", exception.Message);
+        }
+
+        [Theory]
+        [MemberData(nameof(InheritedPersonWithRequiredMembersSetsRequiredMembersWorksAsExpectedSources))]
+        public async Task InheritedPersonWithRequiredMembersSetsRequiredMembersWorksAsExpected(string jsonValue,
+            InheritedPersonWithRequiredMembersSetsRequiredMembers expectedValue)
+        {
+            var options = new JsonSerializerOptions(Serializer.DefaultOptions);
+            options.MakeReadOnly();
+
+            JsonTypeInfo typeInfo = options.GetTypeInfo(typeof(InheritedPersonWithRequiredMembersSetsRequiredMembers));
+            Assert.Equal(3, typeInfo.Properties.Count);
+
+            AssertJsonTypeInfoHasRequiredProperties(GetTypeInfo<InheritedPersonWithRequiredMembersSetsRequiredMembers>(options));
+
+            InheritedPersonWithRequiredMembersSetsRequiredMembers actualValue =
+                await Serializer.DeserializeWrapper<InheritedPersonWithRequiredMembersSetsRequiredMembers>(jsonValue, options);
+            actualValue.Should()
```

Comment:
I don't think one assertions justifies the inclusion of a new third-party library. Assuming you want to check for structural equality you could either:

1. Assert equality of the individual properties
2. Convert the test class to a C# record and change to `Assert.Equal(expectedValue, actualValue);`